content/src/content/jcr_root/apps/core/wcm/components/image/v3/image/clientlibs/ngdmsmartcrop/js/smartcropaction.js

```diff
@@ -0,0 +1,41 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2023 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+/* global CQ */
+(function($, ns, channel, authorNs) {
+    "use strict";
+    ns.image.v3.actions.smartCrop = function() {
+        const editable = this;
+        authorNs.DialogFrame.openDialog(new ns.image.v3.smartCropDialog(editable));
+    };
+
+    ns.image.v3.actions.smartCrop.condition = function(editable) {
+        return authorNs.pageInfoHelper.canModify() && hasNGDMSmartCropAction(editable) && isNGDMImage(editable);
+    };
+
+    function hasNGDMSmartCropAction(editable) {
+        let hasAction = false;
+        editable.config.editConfig.actions.forEach(function(action) {
+            if (typeof action === "object" && (action.name === "ngdm-smartcrop")) {
+                hasAction = true;
```

Comment:
not sure how many iterations would be here, but wouldn't be better to be a normal for loop so you can break once the condition has been met?

Reply:
fixed.